docs/js/features/odata/generic-http-client.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Generic HTTP Client
+sidebar_label: Generic HTTP Client
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- HTTP
+- JavaScript
+- TypeScript
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl'
+
+## What is the Generic HTTP Client
+
+In the [Executing a request using a generated OData client](./execute-odata-request.mdx) section we explained how to make OData requests using the generated OData client.
+This client is the highest layer of convenience and safety for the user.
+Below the OData client there are two further clients: The Generic HTTP client and the Axios client.
+
+<img alt="Destination in CF" src={useBaseUrl('img/odata-client-layers.png')} class="center" />
+
+In this document we explain when and how to use the middle layer directly.
+First we would like to define the boarder between the different clients and what the generic HTTP client is. 
+
+**OData Client:** Provides strong typing with respect to the request (payload,select,filter,...) and response. 
+For each OData service a separate client needs to be generated which defines the types and methods for this service.
+The client takes care of deserializing your request to a URL and JSON payload as well as serializing the response.
+The generated payload and URL are passed to the Generic HTTP Client.
+
+**Generic HTTP Client:** Adds SAP infrastructure specific functionality on top of a standard HTTP Client.
+All OData services use the same generic HTTP client so it contains no service specific information.
+The central aspect of this layer is connectivity based on [destination lookup](../connectivity/destination.md), [On-Premise](../connectivity/on-premise.md) connection via the connectivity service and [web proxy handling](../connectivity/proxy.md).
+In the end all information from the destination, connectivity service and proxy configuration end up in header-fields and [proxy-agents](https://www.npmjs.com/package/proxy-agent).
+These information go one level down to the Axios client.
+
+**Axios HTTP Client:**  Is a widely used open source [HTTP client for node](https://www.npmjs.com/package/axios). 
+This client executes the actual HTTP requests in the end.
+ 
+## When To Use it
+
+In this section we would like to provide some guidance when and when not to use the Generic HTTP client.
+You should consider the Generic HTTP client if:
+- ...You need to use an unsupported feature by the typed OData Client like `upsert` for example. 
+Then you get at least some convenience in the connectivity part.
+- ...You only want to ping a service or trigger a function import without a complicated payload. 
+In such a case the size of a full data model of the typed client is perhaps not worth the benefits. 
+Especially if you ping multiple different services all requiring a data model.
```

Comment:
different services, require same data model? entity? could you please explain a bit?

Reply:
I just removed the whole sentince. The idea that a VDM could be big if you just want to make a very simple call is already said in the sentence before.